RandBLAS/dense.hh

```diff
@@ -183,7 +183,98 @@ DenseSkOp<T,RNG>::~DenseSkOp() {
     }
 }
 
+/** Fill a n_srows \times n_scols submatrix with random values starting at a pointer, from a n_rows \times n_cols random matrix. 
+ * If RandBLAS is compiled with OpenMP threading support enabled, the operation is
+ * parallelized using OMP_NUM_THREADS. The sequence of values genrated is not
+ * dependent on the number of OpenMP threads.
+ *
+ * @tparam T the data type of the matrix
+ * @tparam RNG a random123 CBRNG type
+ * @tparm OP an operator that transforms raw random values into matrix
+ *           elements. See r123ext::uneg11 and r123ext::boxmul.
+ *
+ * @param[in] n_cols the number of columns in the matrix.
+ * @param[in] smat a pointer to a contiguous region of memory with space for
+ *                n_rows \times n_cols elements of type T. This memory will be
+ *                filled with random values.
+ * @param[in] n_srows the number of rows in the submatrix.
+ * @param[in] n_scols the number of colomns in the submatrix.
+ * @param[in] ptr the starting locaiton within the random matrix, for which 
```

Comment:
This is assuming the random matrix is stored in row-major order?

Reply:
Yes. I think the function would also work in column major order if I add an additional argument that takes in the number of rows of the matrix. Should I make those changes? I would have to then do more testing to make sure its correct.

Reply:
The original implementation more or less worked in column-major orientation. If anything, we'd want to switch to entirely column-major. But I suspect that the distinction between these cases isn't significant. Hold off on making any changes on this topic.

Context: when I recently updated RandBLAS to work in terms of "short-axis major" and "long-axis major" for dense sketching operators, I found that it suffices to have _either_ a row-major _or_  a column-major ``fill_rmat`` function, and that we can get the other format by playing with transposes.

Reply:
Okay that sounds good. Please let me know if there is anything else that needs to be addressed.

Reply:
I'll go ahead and merge now. If I run into issues (and there's a decent chance that I will ...), then I'll ping you!